### ISSUE TITLE
Duplicate: Error echoed in footer: session_write_close is being called with 1 parameter

### ISSUE THREAD

**Issue by Julix91** [pantheon-systems/wp-native-php-sessions#144]
**TL; DR ** - Didn't realize MU plugins weren't staying updated and was on `0.6.0` - will update manually, that should fix it.

feel free to close as duplicate of issue #65, solved in version 0.6.1

----

Exact message:

```
Warning: session_write_close() expects exactly 0 parameters, 1 given in /Users/juliansloman/Dev/ffs-mainsite/wp-includes/class-wp-hook.php on line 288
```

Far as I know PHP doesn't do function overloading, so not sure if this is actually breaking anything or not, but I want those errors gone.

Initially this was in staging only (live suppresses warnings, and local didn't get them).
However, commenting out `if (isset($_ENV['PANTHEON_ENVIRONMENT']))` in the `wp-native-php-sessions.php` file in `mu-plugins` allows me to reproduce the issue on local.

Only instance I could find of `session_write_close` being called is here: action `'shutdown'`

![Screen Shot 2020-02-13 at 10 02 28 AM](https://user-images.githubusercontent.com/12877287/74464203-fa777a00-4e47-11ea-8bdf-f02efa29b9fd.png)

```
private function initialize_session_override() {
		session_set_save_handler( '_pantheon_session_open', '_pantheon_session_close', '_pantheon_session_read', '_pantheon_session_write', '_pantheon_session_destroy', '_pantheon_session_garbage_collection' );
		// Close the session before $wpdb destructs itself
		add_action( 'shutdown', 'session_write_close', 999 );
		require_once dirname( __FILE__ ) . '/inc/class-session.php';
	}
```

I'm guessing that's as per 
```
= 0.6.0 (November 23rd, 2016) = 
* Bug fix: Prevents PHP fatal error in `session_write_close()` by running on WordPress' `shutdown` action, before `$wpdb` destructs itself.
```
But I don't see any parameter here... 999 is just the hook priority right? The optional `$accepted_args` is empty, right?

Actually just found this in the change log!
```
= 0.6.1 (May 25th, 2017) =
* Bug fix: Prevents warning session_write_close() expects exactly 0 parameters, 1 given.
```

So seems that this is a regression then?

https://github.com/pantheon-systems/wp-native-php-sessions/pull/65/commits/f6e2b9d60e2d1b79a8a7c77408bb632f262df67d

Seems that a 4th parameter of 0 was added that isn't there anymore - and indeed adding that argument back in makes it work! - Guess the next question would be when/why was it taken back out...

Checked different versions and found even in the last one this fix is in it... Which also explains why I didn't see it in my change log before - because it's not in my change log - *because I'm on an old version!*

https://wordpress.org/support/article/must-use-plugins/

> Plugins in the must-use directory will not appear in the update notifications nor show their update status on the plugins page, so you are responsible for learning about and performing updates on your own.

We don't even use sessions for anything as far as I know. I didn't realize this plugin existed until we ran in to trouble with it... They don't show up under `all` plugins - have their own tab under must use. Interesting. Guess that is because bulk actions available under the `all` tab don't apply to them?

Pantheon can handle core updates, why can't it also include mu-plugin updates - or did it and we somehow missed out on it? Anyway, at least I know how to fix it from here! :)

**Comment by danielbachhuber** [comment]
@Julix91 Glad you were able to track it down!

In the future, one way to make sure a mu plugin stays up to date is to:

1. Install it as a normal plugin.
2. Add a file (e.g. `wp-content/mu-plugins/wp-native-php-sessions.php`) that loads the plugin with a `require_once` statement:

```php
<?php
require_once WP_PLUGIN_DIR . '/wp-native-php-sessions/pantheon-sessions.php';
```

This will make the plugin effectively a mu plugin, while also being able to use WordPress' update system to keep it up to date.
